pkg/ccl/backupccl/file_sst_sink.go

```diff
@@ -350,51 +388,66 @@ func (s *fileSSTSink) copyPointKeys(ctx context.Context, dataSST []byte) error {
 		// bytes, and remove this hacky code.
 		if k.Timestamp.IsEmpty() {
 			if err := s.sst.PutUnversioned(k.Key, valueBuf); err != nil {
-				return err
+				return nil, err
 			}
 		} else {
 			if err := s.sst.PutRawMVCC(k, valueBuf); err != nil {
-				return err
+				return nil, err
 			}
 		}
+		nonEmpty = true
 	}
-	return nil
+	iter.Prev()
+	ok, err := iter.Valid()
+	if err != nil {
+		return nil, err
+	}
+	if !ok {
+		if nonEmpty {
+			return nil, errors.AssertionFailedf("failed to find last key of non-empty file")
+		}
+		return nil, nil
+	}
+	return iter.UnsafeKey().Key.Clone(), nil
 }
 
-func (s *fileSSTSink) copyRangeKeys(dataSST []byte) error {
+func (s *fileSSTSink) copyRangeKeys(dataSST []byte) (roachpb.Key, error) {
 	iterOpts := storage.IterOptions{
 		KeyTypes:   storage.IterKeyTypeRangesOnly,
 		LowerBound: keys.LocalMax,
 		UpperBound: keys.MaxKey,
 	}
 	iter, err := storage.NewMemSSTIterator(dataSST, false, iterOpts)
 	if err != nil {
-		return err
+		return nil, err
 	}
 	defer iter.Close()
-
+	var maxKey roachpb.Key
 	for iter.SeekGE(storage.MVCCKey{Key: keys.MinKey}); ; iter.Next() {
 		if ok, err := iter.Valid(); err != nil {
-			return err
+			return nil, err
 		} else if !ok {
 			break
 		}
 		rangeKeys := iter.RangeKeys()
 		for _, v := range rangeKeys.Versions {
 			rk := rangeKeys.AsRangeKey(v)
+			if rk.EndKey.Compare(maxKey) > 1 {
```

Comment:
Do you think this approach is _wrong_? because I figured this is easier to be obviously correct -- don't need to reason about the start key order implying the end key order -- and there shouldn't be so many that one byte copy into max each time it increases kills us?

Reply:
that's a good point, fine to keep this as is.